### PR TITLE
Refactor 622 text view single type

### PR DESCRIPTION
**Pull request by danielsh28**

### CODE REVIEW

File: src/components/molecules/TextView/TextView.tsx [hunk 1]
```diff
@@ -47,47 +48,39 @@ const useStyles = makeStyles((theme: Theme) => ({
...
-  }
+const getCountsForView = ({
+  items: { severity_fatal_count, severity_light_count, severity_severe_count },
+}: IWidgetCountBySeverityTextData) => {
+  const countsForDesc = Object.entries({
+    severity_fatal_count,
+    severity_light_count,
+    severity_severe_count,
+  }).filter((type) => Boolean(type[1]));
+  return Object.fromEntries(countsForDesc);
```
Comment (yuvalbl) [inline src/components/molecules/TextView/TextView.tsx:59]: please make more readable

Comment (yuvalbl) [reply src/components/molecules/TextView/TextView.tsx:59]:
```
const countBySeverity = {
  fatal: data?.severity_fatal_count || 0,
  severe: data?.severity_severe_count || 0,
  light: data?.severity_light_count || 0,
}
const howManySeverities = [!!countBySeverity.fatal, !!countBySeverity.severe, !!countBySeverity.light ];
const isMultiType = countBySeverity.filter(Boolean).length > 1;

function getSingleType(countBySeverity: CountBySeverity) {
  if(countBySeverity.fatal) {
    return 'fatal'
  }
  ...
}
```

File: src/components/molecules/TextView/TextView.tsx [hunk 2]
```diff
@@ -47,47 +53,45 @@ const useStyles = makeStyles((theme: Theme) => ({
...
+  if (countBySeverity.fatal) {
+    return 'fatal';
   }
-};
+  if (countBySeverity.severe) {
+    return 'sever';
+  }
+  if (countBySeverity.light) {
+    return 'light';
+  }
```
Comment (yuvalbl) [inline src/components/molecules/TextView/TextView.tsx:65]: `return null`

File: src/components/molecules/TextView/TextView.tsx [hunk 3]
```diff
@@ -47,47 +53,45 @@ const useStyles = makeStyles((theme: Theme) => ({
   },
 }));
 
-const extractCount = ({ severity_fatal_count, severity_light_count, severity_severe_count }: any) => {
-  //checking availability of one or more types
-  if ([severity_fatal_count, severity_light_count, severity_severe_count].filter(Boolean).length > 1)
-    return { severity_fatal_count, severity_light_count, severity_severe_count };
-  else {
-    return null;
+function getSingleType(countBySeverity: CountBySeverity): string | undefined {
```
Comment (yuvalbl) [inline src/components/molecules/TextView/TextView.tsx:56]: `string`

File: src/components/molecules/TextView/TextView.tsx [hunk 4]
```diff
@@ -47,47 +53,45 @@ const useStyles = makeStyles((theme: Theme) => ({
...
 
-  const singleType: string | undefined = findSingleType();
-  const headerClass = classNames(classes.headerBase, singleType ? classes.headerSingleType : classes.headerList);
+  const headerClass = classNames(classes.headerBase, isSingleType ? classes.headerSingleType : classes.headerList);
   return (
     <div className={classes.root}>
       <Box className={headerClass} color={brightGreyColor} textAlign="center">
-        <TextViewHeader singleType={singleType} data={data} segmentText={segmentText} />
+        <TextViewHeader
+          singleType={isSingleType && getSingleType(countBySeverity)!}
```
Comment (yuvalbl) [inline src/components/molecules/TextView/TextView.tsx:84]:
```suggestion
          singleType={isSingleType ? getSingleType(countBySeverity) : ''}
```

File: src/components/molecules/TextView/TextView.tsx (hunk 3 above)
Comment (danielsh28) [reply src/components/molecules/TextView/TextView.tsx:56]: done

**Review by eidan66 — APPROVED** [review]

File: src/components/molecules/TextView/TextViewHeader.tsx [hunk 5]
```diff
@@ -9,11 +9,11 @@ import { makeStyles } from '@material-ui/core/styles';
 interface IProps {
   data: IWidgetCountBySeverityTextData;
   segmentText: string;
-  singleType?: string;
+  singleType: string | false;
```
Comment (yuvalbl) [inline src/components/molecules/TextView/TextViewHeader.tsx:12]:
```suggestion
  singleType: string;
```

File: src/components/molecules/TextView/TextViewHeader.tsx [hunk 6]
```diff
@@ -9,11 +9,11 @@ import { makeStyles } from '@material-ui/core/styles';
...
 }
 interface AProps {
   accidentsCount: Number;
-  singleType: string | undefined;
+  singleType: string | false;
```
Comment (yuvalbl) [inline src/components/molecules/TextView/TextViewHeader.tsx:16]:
```suggestion
  singleType: string;
```

File: src/components/molecules/TextView/TextViewHeader.tsx [hunk 7]
```diff
@@ -9,11 +9,11 @@ import { makeStyles } from '@material-ui/core/styles';
 interface IProps {
   data: IWidgetCountBySeverityTextData;
   segmentText: string;
-  singleType?: string;
+  singleType: string | false;
 }
 interface AProps {
```
Comment (yuvalbl) [inline src/components/molecules/TextView/TextViewHeader.tsx:14]: `AProps` ?

**Pull request merged**